### ISSUE TITLE
pdbg -a threadstatus is showing .S. for all the cores

### ISSUE THREAD

**Issue by PavamanSubramaniyam** [open-power/pdbg#39]
I have followed the steps mentioned in Cross compiling for BMC (ARM) of the README for Building on a x86 server and then did rsync of the binary to the Witherspoon BMC.

Then executed the ./pdbg -a threadstatus command in the Witherspoon BMC shell.

```
root@witherspoon:~# ./pdbg -a threadstatus

p0t:   0   1   2   3
c01:  .S. .S. .S. .S.
c02:  .S. .S. .S. .S.
c03:  .S. .S. .S. .S.
c04:  .S. .S. .S. .S.
c05:  .S. .S. .S. .S.
c06:  .S. .S. .S. .S.
c07:  .S. .S. .S. .S.
c08:  .S. .S. .S. .S.
c10:  .S. .S. .S. .S.
c11:  .S. .S. .S. .S.
c12:  .S. .S. .S. .S.
c13:  .S. .S. .S. .S.
c14:  .S. .S. .S. .S.
c15:  .S. .S. .S. .S.
c16:  .S. .S. .S. .S.
c17:  .S. .S. .S. .S.
c18:  .S. .S. .S. .S.
c19:  .S. .S. .S. .S.
c20:  .S. .S. .S. .S.
c21:  .S. .S. .S. .S.
c22:  .S. .S. .S. .S.
c23:  .S. .S. .S. .S.

p1t:   0   1   2   3
c02:  .S. .S. .S. .S.
c03:  .S. .S. .S. .S.
c04:  .S. .S. .S. .S.
c05:  .S. .S. .S. .S.
c06:  .S. .S. .S. .S.
c07:  .S. .S. .S. .S.
c08:  .S. .S. .S. .S.
c09:  .S. .S. .S. .S.
c10:  .S. .S. .S. .S.
c11:  .S. .S. .S. .S.
c12:  .S. .S. .S. .S.
c13:  .S. .S. .S. .S.
c14:  .S. .S. .S. .S.
c15:  .S. .S. .S. .S.
c16:  .S. .S. .S. .S.
c17:  .S. .S. .S. .S.
c18:  .S. .S. .S. .S.
c19:  .S. .S. .S. .S.
c20:  .S. .S. .S. .S.
c21:  .S. .S. .S. .S.
c22:  .S. .S. .S. .S.
c23:  .S. .S. .S. .S.
```

But in the README, under the Get thread status section the output is shown as "A A A A" for each of the cores.
So, is the output we are getting on Witherspoon BMC which shows as "c23:  .S. .S. .S. .S." for thread status is expected.

**Comment by liyi-ibm** [comment]
Seems to me this is not a bug. If you set kernel option to "powersave=off", Or,  if in the OS you run "ppc64_cpu --frequency" to make all cores busy, the threadstate will be "A".

**Comment by PavamanSubramaniyam** [comment]
The Linux Host got hung and is not responsive. So I wanted to debug the host unresponsiveness using the pdbg tool from the BMC. 
Do you infer here that ".S." is expected output for the thread state, when the linux host went into hang state.

**Comment by liyi-ibm** [comment]
@PavamanSubramaniyam, As I observed, if cpu is idle (that is, OS booted and there is no workload), 'pdbg -a thread status" shows the thread status is "S".

**Comment by apopple** [comment]
@PavamanSubramaniyam Yes, ".S." is the expected output. We need to update the documentation to reflect this. If the thread status is "S" it means the thread is sleeping/inactive which from the Linux perspective means it's reached and isn't running anything (ie. not very interesting for debugging). As @liyi-ibm it is generally easier to debug systems by booting with powersave="off".
